samples/jsonParserComparison.js

```diff
@@ -0,0 +1,165 @@
+  const dropTempTable = `drop table if exists ${testVariantTempName}`;
+
+  const insertVariant = (tableName)=> `insert into ${tableName}
+                         select parse_json(value)
+                         from ${testVariantTempName}`;
+  const selectCountVariant = (tableName) => `select count(colA) from ${(tableName)}`;
+
+  const testCases = [];
+  if (!process.argv[4] || process.argv[4].toString().includes('Function')) {
+    testCases.push({parser: 'Function', jsonColumnVariantParser: (rawColumnValue) => new Function(`return (${rawColumnValue})`)});
+  }
+  if (!process.argv[4] || process.argv[4].toString().includes('betterEval')) {
+    testCases.push({parser: 'betterEval', jsonColumnVariantParser: (rawColumnValue) => require('better-eval').call('(' + rawColumnValue + ')')});
+  }
+  if (!process.argv[4] || process.argv[4].toString().includes('vm')) {
+    testCases.push({parser: 'vm', jsonColumnVariantParser: rawColumnValue => require('vm').runInNewContext('(' + rawColumnValue + ')')});
+  }
+  // eval contain vulnerability so we decide to resign using it
+  // if (!process.argv[4] || process.argv[4].toString().contains('eval')) {
+  //   testCases.push({parser: 'eval', jsonColumnVariantParser: rawColumnValue => eval('(' + rawColumnValue + ')')})
+  // };
+  if (!process.argv[4] || process.argv[4].toString().includes('JSON')) {
+    testCases.push({parser: 'JSON', jsonColumnVariantParser: rawColumnValue => JSON.parse(rawColumnValue)});
+  }
+
+  const execute = ({parser, jsonColumnVariantParser}) => {
+    console.log(`\nTesting for parser:  ${parser}`);
+    const testVariantTableName = `testVariantTable000${parser}`;
+
+    return new Promise(async (resolve, reject) => {
+      snowflake.configure({
+        jsonColumnVariantParser: jsonColumnVariantParser
+      });
+
+      await helpers.executeQuery(connection, createTempTableWithJsonData);
+      await helpers.executeQuery(connection, createTableWithVariant(testVariantTableName));
+      await helpers.executeQuery(connection, insertVariant(testVariantTableName));
+      await helpers.executeQuery(connection, selectCountVariant(testVariantTableName));
+
+      const queryTimeLabel = parser + 'SelectTime';
+      let avgBlock = 0, minBlock = 999999999999999, maxBlock = 0;
+      let blockCount = 0;
+      blocked((time) => {
+        blockCount++;
+        avgBlock += time;
+        minBlock = minBlock > time ? time : minBlock;
+        maxBlock = maxBlock < time ? time : maxBlock;
+      });
+
+      console.time(queryTimeLabel);
+      let count = 0;
+      const streamResult = true;
+      connection.execute({
+        streamResult: streamResult,
+        sqlText: `select *
+                  from IDENTIFIER(?) LIMIT ${selectLimit}`,
+        binds: [testVariantTableName],
+        complete: function (err, stmt) {
+          const stream = stmt.streamRows();
+          stream.on('readable', function () {
```

Comment:
let's add second test suite for reading events data/error/end

Reply:
added